File: app/src/main/java/com/devvikram/networkutil/MainActivity.java
package com.devvikram.networkutil;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

import com.devvikram.networklib.RegisterNetwork;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        RegisterNetwork.registerNetworks(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        RegisterNetwork.unRegisterNetwork(this);
    }
}

File: networkcheck/src/main/java/com/devvikram/networkcheck/MainActivity.kt
package com.devvikram.networkcheck

import android.content.IntentFilter
import android.net.ConnectivityManager
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private val receiver = NetworkAndInterChangeReceiver()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val filter = IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION)
        registerReceiver(receiver, filter)
    }

    override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(receiver)
    }
}